ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitter.scala

```diff
@@ -185,52 +185,57 @@ private[kvutils] class TransactionCommitter(
         getCurrentConfiguration(defaultConfig, commitContext, logger)
       val timeModel = config.timeModel
 
-      commitContext.recordTime match {
-        case Some(recordTime) =>
-          val givenLedgerTime = transactionEntry.ledgerEffectiveTime.toInstant
+      val maybeDeduplicateUntil =
+        getLedgerDeduplicateUntil(transactionEntry, commitContext)
+      val minimumRecordTime = transactionMinRecordTime(
+        transactionEntry.submissionTime.toInstant,
+        transactionEntry.ledgerEffectiveTime.toInstant,
+        maybeDeduplicateUntil,
+        timeModel,
+      )
+      val maximumRecordTime = transactionMaxRecordTime(
+        transactionEntry.submissionTime.toInstant,
+        transactionEntry.ledgerEffectiveTime.toInstant,
+        timeModel,
+      )
+      commitContext.deduplicateUntil = maybeDeduplicateUntil
+      commitContext.minimumRecordTime = Some(minimumRecordTime)
+      commitContext.maximumRecordTime = Some(maximumRecordTime)
+      val outOfTimeBoundsLogEntry = DamlLogEntry.newBuilder
+        .setTransactionRejectionEntry(
+          buildRejectionLogEntry(
+            transactionEntry,
+            RejectionReason.InvalidLedgerTime(
+              s"Record time is outside of valid range [$minimumRecordTime, $maximumRecordTime]"
+            ),
+          )
+        )
+        .build
+      commitContext.outOfTimeBoundsLogEntry = Some(outOfTimeBoundsLogEntry)
```

Comment:
This is work you only need to do for pre-execution and it feels wasteful to do it anyway.

Reply:
You're right, but we're planning on deleting the other code path at some point soon anyway, so I didn't worry about it too much.